elasticsearch-client/elasticsearch-client-v6/src/main/java/fr/pilato/elasticsearch/crawler/fs/client/v6/ElasticsearchClientV6.java

```diff
@@ -374,6 +381,25 @@ public String getDefaultTypeName() {
         return INDEX_TYPE_DOC;
     }
 
+    private static TrustManager[] trustAllCerts = new TrustManager[]{new X509TrustManager() {
```

Comment:
To avoid code duplication for standard things like this one, could you move this code to `ElasticsearchClientUtil` class?

And create inside a method like:

```java
public SchemeIOSessionStrategy generateNoCheckSSLStrategy() {
                    SSLContext sc;
                    try {
                        sc = SSLContext.getInstance("SSL");
                        sc.init(null, trustAllCerts, new SecureRandom());
                    } catch (KeyManagementException | NoSuchAlgorithmException e) {
                        logger.warn("Failed to get SSL Context", e);
                        throw new RuntimeException(e);
                    }
                    return new SSLIOSessionStrategy(sc, new NullHostNameVerifier());
}
```

And use that later?

Reply:
It looks like the `ElasticsearchClientUtil ` class is used to construct the ESClient, can we move the codes into that one? or you are saying the interface `ElasticsearchClient`

Reply:
I'm meaning to share the same piece of code which builds `SchemeIOSessionStrategy` instance in `ElasticsearchClientUtil` class. And call that method from `ElasticsearchClientV6` and `ElasticsearchClientV7` classes as you did.